packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -734,9 +734,14 @@ abstract class State<T extends StatefulWidget> {
   /// [initState]. If the parent updates this location in the tree to a new
   /// widget with the same [runtimeType] and [key] as the current configuration,
   /// the framework will update this property to refer to the new widget and
-  /// then call [didUpdateConfig], passing the old configuration as an argument.
-  T get config => _config;
-  T _config;
+  /// then call [didUpdateWidget], passing the old configuration as an argument.
+  T get widget => _widget;
+  T _widget;
+
+  /// Deprecated getter for [widget]. Use the [widget] getter instead. This
+  /// getter will be removed on 2017/04/20.
+  @Deprecated("Will be removed 2017/04/20")
```

Comment:
I recommend the following message instead:
`@Deprecated('The "config" property is now called "widget". Please migrate now, as we will remove this feature soon.')`

...or just remove it. It doesn't make any real difference if we cause the apocalypse today or on the 20th.